src/components/graph-view.js

```diff
@@ -692,20 +692,22 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
       const deltaX = position.x - node.x;
       const deltaY = position.y - node.y;
 
-      const nodesToMove = selectedNodes.find(n => n[nodeKey] === node[nodeKey])
+      const isSelected = selectedNodes.find(n => n[nodeKey] === node[nodeKey]);
+
+      const nodesToMove = isSelected
         ? selectedNodes
         : selectedNodes.concat([node]);
 
       GraphUtils.yieldingLoop(nodesToMove.length, 50, i => {
         const node = nodesToMove[i];
         const nodeMapNode = this.getNodeById(node[nodeKey]);
 
-        // node moved
+        // move node before rendering edges
         node.x += deltaX;
         node.y += deltaY;
 
         this.renderConnectedEdgesFromNode(nodeMapNode, true);
```

Comment:
I am wondering the idea was avoid multiple calls, so if remove  `this.renderConnectedEdgesFromNode(nodeMapNode, true)` on line 709 and keep line 710 as is `this.asyncRenderNode(node)`

Reply:
This is the same concern that @hodos added above -- see my response there! :)